desktop/extensions-sdk/build/set-up/frontend-extension-tutorial.md

```diff
@@ -0,0 +1,309 @@
+---
+title: Set up an advanced frontend extension
+description: Advanced frontend extension tutorial
+keywords: Docker, extensions, sdk, build
+redirect_from:
+  - /desktop/extensions-sdk/tutorials/react-extension/
+  - /desktop/extensions-sdk/build/set-up/react-extension/
+  - /desktop/extensions-sdk/build/set-up/minimal-frontend-using-docker-cli/
+---
+
+To start creating your extension, you first need a directory with files which range from the extension’s source code to the required extension-specific files. This page provides information on how to set up a simple Docker extension that contains only a UI part.
+
+> Note
+>
+> Before you start, make sure you have installed the latest version of [Docker Desktop](../../../release-notes.md).
+
+## Extension folder structure
+
+The easiest way to create a new extension is to run `docker extension init <my-extension>` as in the [Quickstart](../../quickstart.md).
+This will create a new directory `<my-extension>` that contains a fully functional extension.
+
+> **Note**
+>
+> The `docker extension init` generates a React based extension. But you can still use it as a starting point for 
+> your own extension and use any other frontend framework, like Vue, Angular, Svelte, etc. or event stay with 
+> vanilla Javascript.
+
+Although you can start from an empty directory of from the `react-extension` [sample folder](https://github.com/docker/extensions-sdk/tree/main/samples){:target="_blank" rel="noopener" class="_"},
+it's highly recommended that you start from the `docker extension init` command and change it so to suit your needs.
+
+```bash
+.
+├── Dockerfile # (1)
+├── ui # (2)
+│   ├── public # (3)
+│   │   └── index.html
+│   ├── src # (4)
+│   │   ├── App.tsx
+│   │   ├── index.tsx
+│   ├── package.json
+│   └── package-lock.lock
+│   ├── tsconfig.json
+├── docker.svg # (5)
+└── metadata.json # (6)
+```
+
+1. Contains everything required to build the extension and run it in Docker Desktop.
+2. High-level folder containing your front-end app source code.
+3. Assets that aren’t compiled or dynamically generated are stored here. These can be static assets like logos or the robots.txt file.
+4. The src, or source folder contains all the React components, external CSS files, and dynamic assets that are brought into the component files.
+5. The icon that is displayed in the left-menu of the Docker Desktop Dashboard.
+6. A file that provides information about the extension such as the name, description, and version.
+
+## Create a Dockerfile
```

Comment:
If users have run `docker extension init`, then this title is misleading (the Dockerfile has already been created for them).

```suggestion
## Adapting the Dockerfile (optional)
```